docs/rollup-boost-ha.md

```diff
@@ -0,0 +1,245 @@
+    Enabled,
+    /// Forward Engine API requests to the builder, validate builder payloads but
+    /// fallback to default execution payload
+    DryRun,
+    // Do not forward Engine API requests to the builder 
+    Disabled,
+}
+```
+
+<br>
+
+## Debug API
+
+`rollup-boost` exposes a Debug API that allows operators to inspect and modify the current execution mode at runtime without restarting the service. This provides flexibility to dynamically enable, disable, or dry-run external block production based on builder behavior or network conditions. The Debug API is served over HTTP using JSON RPC and consists of the following endpoints:
+
+### `debug_setExecutionMode`
+
+Sets the current execution mode for `rollup-boost`.
+
+**Request**:
+
+```
+{
+  "method": "debug_setExecutionMode",
+  "params": [ "enabled" | "dry_run" | "disabled" ],
+  "id": 1,
+  "jsonrpc": "2.0"
+}
+```
+
+**Response**:
+
+```
+{
+  "result": null,
+  "id": 1,
+  "jsonrpc": "2.0"
+}
+```
+
+### `debug_getExecutionMode`
+
+Retrieves the current execution mode.
+
+**Request**:
+
+```
+{
+  "method": "debug_getExecutionMode",
+  "params": [],
+  "id": 1,
+  "jsonrpc": "2.0"
+}
+```
+
+**Response:**
+
+```
+{
+  "result": "enabled" | "dry_run" | "disabled",
+  "id": 1,
+  "jsonrpc": "2.0"
+}
+```
+
+<br>
+
+## Failure Scenarios
+
+Below is a high level summary of how each failure scenario is handled. All existing failure modes assumed by upstream `op-conductor` are maintained:
+
+| Failure Scenario | Category | Scenario and Solution |
+| --- | --- | --- |
+| Leader Sequencer Fails | Sequencer Failure | Conductor will detect sequencer failure and start to transfer leadership to another node, which will start sequencing instead. `op-conductor` will prioritize electing a leader with full health however if there are no sequencers that are fully healthy, a sequencer with partial health will be selected. Once the originally failed sequencer is back online, it will join the sequencer set as a follower. |
+| Leader `rollup-boost` Unhealthy | Rollup Boost Failure | Leader sequencer `rollup-boost` becomes unhealthy, causing `op-conductor`s sequencer health checks to fail. If `op-conductor` is configured to failover in the event that `rollup-boost` is partially healthy conductor will elect a new leader. This failure mode is the same as a typical leader sequencer failure. Once the sequencer recovers, it will continue to participate in the cluster as a follower. If `op-conductor` is configured not to failover in the event of `rollup-boost` partial health, the sequencer will continue to publish the default exectuion client's blocks until the builder has recovered back to full health. |
+| Follower `rollup-boost` Unhealthy | Rollup Boost Failure | Follower sequencer `rollup-boost` becomes unhealthy. The leader sequencer is unaffected. Once the sequencer recovers, it will continue to participate in the cluster as a follower.|
```

Comment:
same here

Reply:
I responded to the prev comment but copying here for visibility. 

Failure scenarios have been updated clarify behavior for both partial and full `rollup-boost` failure as well as local execution client and builder failures.